ql/src/java/org/apache/hadoop/hive/ql/parse/repl/load/message/TruncatePartitionHandler.java

```diff
@@ -35,17 +39,39 @@
 public class TruncatePartitionHandler extends AbstractMessageHandler {
   @Override
   public List<Task<?>> handle(Context context) throws SemanticException {
-    AlterPartitionMessage msg = deserializer.getAlterPartitionMessage(context.dmd.getPayload());
-    final TableName tName = TableName.fromString(msg.getTable(), null,
-        context.isDbNameEmpty() ? msg.getDB() : context.dbName);
-
-    Map<String, String> partSpec = new LinkedHashMap<>();
+    final TableName tName;
     org.apache.hadoop.hive.metastore.api.Table tblObj;
     try {
-      tblObj = msg.getTableObj();
-      Iterator<String> afterIterator = msg.getPtnObjAfter().getValuesIterator();
-      for (FieldSchema fs : tblObj.getPartitionKeys()) {
-        partSpec.put(fs.getName(), afterIterator.next());
+      if (MetastoreConf.getBoolVar(context.hiveConf,
+          MetastoreConf.ConfVars.NOTIFICATION_ALTER_PARTITIONS_V2_ENABLED)) {
+        AlterPartitionsMessage singleMsg = deserializer.getAlterPartitionsMessage(
+            context.dmd.getPayload());
+        tName = TableName.fromString(singleMsg.getTable(), null,
+            context.isDbNameEmpty() ? singleMsg.getDB() : context.dbName);
+        tblObj = singleMsg.getTableObj();
```

Comment:
nit: can we create the `tName` from the `tblObj`?

Reply:
Replication tests fail because during the replication it would require source DB name(which we get it from 'context' variable).